### PR TITLE
ttH 13.6 TeV Signal Sample

### PR DESCRIPTION
**Pull request by mobassirameen**
Dear expert,

We have produced gridpack placed at path [1] using the input process cards [2] for signal samples at 13.6 TeV for RunIII production. Only the beam energy is changed in the input card.
Production of gridpack is followed by the twiki [3]. But following commands mentioned in twiki page gives me following information about missing files:
```shell
Start of job on  Wed Jun 7 09:42:02 CEST 2023
LHAPDF_BASE is set to: /cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/lhapdf/6.4.0-68defff11ffd434c73727d03802bfb85
/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg
I am here:
/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./powheg.input': No such file or directory
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./JHUGen.input': No such file or directory
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./*.dat': No such file or directory
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./pwhg_main': No such file or directory
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./*.grid': No such file or directory
cp: cannot stat '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./*.cdf': No such file or directory
cp: '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./check_dag_success.py' and './check_dag_success.py' are the same file
cp: '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./make_rwl.py' and './make_rwl.py' are the same file
cp: '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./run_pwg_condor.py' and './run_pwg_condor.py' are the same file
cp: '/afs/cern.ch/user/m/moameen/ttHRun3MCSamples/gridpackProd/CMSSW_12_4_8/src/genproductions/bin/Powheg/./run_pwg_parallel_condor.py' and './run_pwg_parallel_condor.py' are the same file
Process is: ttH
``` 
But we produced gridpack ignoring the JHUGen.input file and similarly mentioned above.Please review them, if there are issues ignoring them, do we need to cinsider this file for the ttH process production, if yes, please let us know.

 Mohammad Mobassir Ameen
 (ttH+HH MC contact)
 
Adding[ @hftsoi](https://github.com/hftsoi)[ @attikis](https://github.com/attikis) 
[1] /afs/cern.ch/user/m/moameen/public
[2] https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/Run3/13p6TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125.input
 [3] https://twiki.cern.ch/twiki/bin/viewauth/CMS/PowhegBOXPrecompiled

### CODE REVIEW

**Comment by attikis** [comment]
Hi all, 

a kind reminder for this pending PR. 

Thank you,
Alexandros & Ho-Fung for HIG MC

**Comment by vischia** [comment]
(following)

**Comment by menglu21** [comment]
Hi, I went through the script for powheg, those warnings should be fine. @agrohsje @covarell are those lines necessary for condor submission in oneshot run https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/Templates/prepareJob_template.sh#L35-L49

**Comment by mobassirameen** [comment]
Hi @menglu21 The purpose of this PR was to review the card and to merged. Sorry for the late reply.

**Comment by covarell** [comment]
Hi, the missing files are not a concern, if the gridpack can run and produce events correctly.

However, files in this commit (not related to POWHEG) are useless and should not be merged in my opinion.

File: genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M-125_TuneCP5_13p6TeV_powheg-pythia8.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    outputFile = cms.string('cmsgrid_final.lhe'),
+    scriptName = cms.FileInPath('GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh')
+)
+
+#Link to datacards:
+# For Run-2 https://github.com/cms-sw/genproductions/blob/0a0040eb78ea5cafe8287837d333869e2a676d3d/bin/Powheg/production/2017/13TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13TeV_M125/ttH_inclusive_hdamp_NNPDF31_13TeV_M125.input
+# For Run-3: https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/Run3/13p6TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125.input
+
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.MCTunes2017.PythiaCP5Settings_cfi import *
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M-125_TuneCP5_13p6TeV_powheg-pythia8.py:17]: it should be 13.6TeV one

File: genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M-125_TuneCP5_13p6TeV_powheg-pythia8.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+from Configuration.Generator.Pythia8PowhegEmissionVetoSettings_cfi import *
+from Configuration.Generator.PSweightsPythia.PythiaPSweightsSettings_cfi import *
+
+generator = cms.EDFilter("Pythia8HadronizerFilter",
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M-125_TuneCP5_13p6TeV_powheg-pythia8.py:21]: it should be the concurrent one

File: genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M125_TuneCP5_13p6TeV_powheg-pythia8.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    outputFile = cms.string('cmsgrid_final.lhe'),
+    scriptName = cms.FileInPath('GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh')
+)
+
+#Link to datacards:
+# For Run-2: https://github.com/cms-sw/genproductions/blob/0a0040eb78ea5cafe8287837d333869e2a676d3d/bin/Powheg/production/2017/13TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13TeV_M125/ttH_inclusive_hdamp_NNPDF31_13TeV_M125.input
+# For Run-3: https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/Run3/13p6TeV/Higgs/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125/ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125.input
+
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.MCTunes2017.PythiaCP5Settings_cfi import *
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M125_TuneCP5_13p6TeV_powheg-pythia8.py:16]: same comment here

File: genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M125_TuneCP5_13p6TeV_powheg-pythia8.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+from Configuration.Generator.Pythia8PowhegEmissionVetoSettings_cfi import *
+from Configuration.Generator.PSweightsPythia.PythiaPSweightsSettings_cfi import *
+
+generator = cms.EDFilter("Pythia8HadronizerFilter",
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/TTHtoNon2B_M125_TuneCP5_13p6TeV_powheg-pythia8.py:20]: same comment here

**Review by menglu21 — COMMENTED** [review]
@mobassirameen please remove the test.txt and mytext.txt. and the two fragments are the same, right, if yes please only keep one of them and address the comments inline

**Comment by menglu21** [comment]
> Hi, the missing files are not a concern, if the gridpack can run and produce events correctly.
> 
> However, files in this commit (not related to POWHEG) are useless and should not be merged in my opinion.

@covarell thanks for your confirmation, I agree with you on the useless files.

**Comment by mobassirameen** [comment]
Hi @menglu21, @covarell
Thanks for your comments and suggestions. I will correct it and will let you know.

**Comment by mobassirameen** [comment]
Hi @menglu21 @covarell 
Please have a look on my commits, I have changed according to your suggestions and also can find powheg input card.

**Comment by mobassirameen** [comment]
> Hi @mobassirameen could you please modify the pythiahadronization to the concurrent one

**Comment by mobassirameen** [comment]
Hi @menglu21 I have modify the pythiahadronization to the concurrent one. Please check

**Comment by mobassirameen** [comment]
Hi @menglu21 
I want to bring attention to this PR to review the following Powheg card:
ttH_inclusive_hdamp_NNPDF31_13p6TeV_M125.input
It is been long time. So, I request you please check!

**Comment by attikis** [comment]
hi @menglu21, 

a kind reminder to review this PR now that Pythia8ConcurrentHadronizerFilter has been added.
Please let us know if you require other actions or clarifications.

Thank you,
Alexandros & Ho-Fung for HIG MC

**Pull request merged**